packages/flutter/lib/src/widgets/basic.dart

```diff
@@ -2265,6 +2267,7 @@ class AssetImage extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
     return new RawImageResource(
+      key: new ObjectKey(name),
```

Comment:
This should be the same as NetworkImage, i.e. make the key be based on the ImageResource.